File: java7/src/main/java/com/thoughtworks/jcprogram/functional/solution/StringReducer.java
package com.thoughtworks.jcprogram.functional.solution;

import java.util.List;

import static java.util.Arrays.asList;

// Given a list of strings like {"Bill", "Archer", "Lana"}
// Print the first letter of each of the string that have a length of 4.
// For example, "BL"

public class StringReducer {

    public static void main(String[] args) {
        List<String> strings = asList("Bill", "Archer", "Lana");
        System.out.println(new StringReducer().reduceToFirstLetterOfLengthFourStrings(strings));
    }

    public String reduceToFirstLetterOfLengthFourStrings(List<String> strings) {
        String result = "";
        for (String string : strings) {
            if (string.length() == 4){
                Character c = string.charAt(0);
                result += c.toString();
            }
        }
        return result;
    }
}

File: scala/src/main/scala/com/thoughtworks/jcprogram/functional/exercises/StringReducer.scala
package com.thoughtworks.jcprogram.functional.exercises

// Given a list of strings like {"Bill", "Archer", "Lana"}
// Print the first letter of each of the string that have a length of 4.
// For example, "BL"

object StringReducer {

  def reduceToFirstLetterOfLengthFourStrings(strings: Set[String]){

  }

  def main(args: Array[String]) {
    val strings = Set("Bill", "Archer", "Lana")

    val reducer = StringReducer
    reducer.reduceToFirstLetterOfLengthFourStrings(strings)
  }
}
